app/models/registry/auth_scope.rb

```diff
@@ -0,0 +1,48 @@
+# Registry::AuthScope parses the scope string so it can be used afterwards for
+# the "registry" type.
+class Registry::AuthScope
+  # The given Registry was not found.
+  class ResourceIsNotFound < StandardError; end
+
+  attr_accessor :resource, :actions, :resource_type, :resource_name
+
+  def initialize(registry, scope_string)
+    @scope_string = scope_string
+    @registry     = registry
+    parse_scope_string!
+  end
+
+  # Returns the registry required by this scope.
+  def resource
+    reg = Registry.find_by(hostname: @registry.hostname)
+    if reg.nil?
+      Rails.logger.warn "Could not find registry #{@registry.hostname}"
+      raise ResourceIsNotFound
+    end
+    reg
+  end
+
+  # Returns an array containing the scopes available for this registry object.
+  # It returns a ["*"] if it's an invalid scope, because this will not be
+  # implemented as a method.
+  def scopes
```

Comment:
Can you provide more details about the two scopes, like when they are used? This is not so clear right now as it is for the namespace scope (where pull and push are pretty obvious). The goal is to avoid someone else to dig into the distribution code later on ;)